shell/platform/windows/flutter_windows.cc

```diff
@@ -0,0 +1,270 @@
+  std::vector<const char*> argv = {"placeholder"};
+  if (arguments_count > 0) {
+    argv.insert(argv.end(), &arguments[0], &arguments[arguments_count]);
+  }
+
+  window->CreateRenderSurface();
+
+  FlutterRendererConfig config = {};
+
+  // Provide the necessary callbacks for rendering within a win32 window.
+  config.type = kOpenGL;
+  config.open_gl.struct_size = sizeof(config.open_gl);
+  config.open_gl.make_current = [](void* userData) -> bool {
+    auto host = static_cast<flutter::FlutterWindow*>(userData);
+    return host->MakeCurrent();
+  };
+  config.open_gl.clear_current = [](void* userData) -> bool {
+    auto host = static_cast<flutter::FlutterWindow*>(userData);
+    return host->ClearContext();
+  };
+  config.open_gl.present = [](void* userData) -> bool {
+    auto host = static_cast<flutter::FlutterWindow*>(userData);
+    return host->SwapBuffers();
+  };
+  config.open_gl.fbo_callback = [](void* userData) -> uint32_t { return 0; };
+  config.open_gl.gl_proc_resolver = [](void* userData,
+                                       const char* what) -> void* {
+    return eglGetProcAddress(what);
+  };
+
+  FlutterProjectArgs args = {};
+  args.struct_size = sizeof(FlutterProjectArgs);
+  args.assets_path = assets_path;
+  args.icu_data_path = icu_data_path;
+  args.command_line_argc = static_cast<int>(argv.size());
+  args.command_line_argv = &argv[0];
+  args.platform_message_callback = WindowsOnFlutterPlatformMessage;
+  // args.custom_task_runners = custom_task_runners; TODO
+
+  FlutterEngine engine = nullptr;
+  auto result =
+      FlutterEngineRun(FLUTTER_ENGINE_VERSION, &config, &args, window, &engine);
+  if (result != kSuccess || engine == nullptr) {
+    std::cerr << "Failed to start Flutter engine: error " << result
+              << std::endl;
+    return nullptr;
+  }
+
+  return engine;
+}
+
+bool FlutterDesktopInit() {
+  return true;
+}
+
+void FlutterDesktopTerminate() {}
+
+FlutterDesktopWindowControllerRef FlutterDesktopCreateWindow(
+    int initial_width,
+    int initial_height,
+    const char* title,
+    const char* assets_path,
+    const char* icu_data_path,
+    const char** arguments,
+    size_t argument_count) {
+  auto win = std::make_unique<flutter::FlutterWindow>(
+      title, 10, 10, initial_width, initial_height);
+
+  flutter::FlutterWindow* window = win.get();
+
+  auto engine = RunFlutterEngine(window, assets_path, icu_data_path, arguments,
+                                 argument_count);
+
+  if (engine == nullptr) {
+    return nullptr;
+  }
+
+  auto state = window->SetState(engine);
+  state->window = std::move(win);
+
+  // Trigger an initial size callback to send size information to Flutter.
+  window->OnResize();
+
+  return state;
+}
+
+void FlutterDesktopDestroyWindow(FlutterDesktopWindowControllerRef controller) {
+  FlutterEngineShutdown(controller->engine);
+  delete controller;
+}
+
+void FlutterDesktopWindowSetHoverEnabled(FlutterDesktopWindowRef flutter_window,
```

Comment:
I would rather have these APIs not present in the initial version than present but silently non-functional.

Reply:
Ignore; per earlier comment I now understand the benefit of landing as-is.